extensions/amp-analytics/0.1/test/test-variables.js

```diff
@@ -127,56 +129,66 @@ describe('amp-analytics.VariableService', function() {
         new ExpansionOptions({'foo': ' Hello world! '}))
         .then(actual => expect(actual).to.equal('')));
 
-  describe('filter:', () => {
-    const vars = new ExpansionOptions({'foo': ' Hello world! '});
+  describes.fakeWin('filters', {amp: true}, env => {
+    let ampdoc;
+    let urlReplacementService;
 
     beforeEach(() => {
-      sandbox.stub(variables, 'isFilterExperimentOn_').callsFake(() => true);
+      toggleExperiment(env.win, 'url-replacement-v2', true);
+      ampdoc = env.ampdoc;
+      urlReplacementService = Services.urlReplacementsForDoc(ampdoc);
+    });
+
+    afterEach(() => {
+      toggleExperiment(env.win, 'url-replacement-v2');
     });
 
     function check(input, output) {
-      return variables.expandTemplate(input, vars).then(actual =>
-        expect(actual).to.equal(output));
+      const filters = variables.getFilters();
+      const expanded = urlReplacementService.expandAsync(input, filters);
+      return expect(expanded).to.eventually.equal(output);
     }
 
-    it('default works', () => check('${bar|default:baz}', 'baz'));
+    it('default works', () => check('DEFAULT(one,two)', 'one'));
+    it('default works without first arg', () => check('DEFAULT(,two)', 'two'));
```

Comment:
another one that default works with first arg length 0?

Reply:
done